hydradx/model/indexer_utils.py

```diff
@@ -390,6 +397,152 @@ def get_fee_pcts(data, asset_id):
     return fee_pcts
 
 
+def get_current_stableswap_pools(asset_info: dict[int: AssetInfo] = None):
+    stableswap_data = {
+        pool: get_latest_stableswap_data(pool)
+        for pool in get_stablepool_ids()
+    }
+    if asset_info is None:
+        asset_info = get_asset_info_by_ids()
+    stableswap_pools = []
+    for pool in stableswap_data.values():
+        if min(pool['liquidity'].values()) > 0:
+            stableswap_pools.append(
+                StableSwapPoolState(
+                    tokens={asset_info[tkn_id].symbol: pool['liquidity'][tkn_id] for tkn_id in pool['liquidity']},
+                    amplification=pool['finalAmplification'],
+                    trade_fee=pool['fee'],
+                    unique_id=asset_info[pool['pool_id']].name,
+                )
+            )
+    return stableswap_pools
+
+def get_current_omnipool():
+    asset_info = get_asset_info_by_ids()
+    for asset in asset_info.values():
+        if asset.asset_type == 'StableSwap':
+            asset.symbol = asset.name
+    current_block = get_current_block_height()
+    omnipool_data = get_omnipool_asset_data(min_block_id=current_block - 10000, max_block_id=current_block)
+    liquidity = {}
+    lrna = {}
+    for item in reversed(omnipool_data):
+        if asset_info[item['assetId']].symbol not in liquidity:
+            liquidity[asset_info[item['assetId']].symbol] = int(item['balances']['free']) / (10 ** asset_info[item['assetId']].decimals)
+            lrna[asset_info[item['assetId']].symbol] = int(item['assetState']['hubReserve']) / (10 ** asset_info[1].decimals)
+        if len(liquidity) == len(asset_info):
+            break
+    asset_fee, lrna_fee = get_current_omnipool_fees(asset_info)
+    for tkn in liquidity:
+        if tkn not in asset_fee.current:
+            asset_fee.current[tkn] = 0.0
+            asset_fee.last_updated[tkn] = current_block
+        if tkn not in lrna_fee.current:
+            lrna_fee.current[tkn] = 0.0
+            lrna_fee.last_updated[tkn] = current_block
+    omnipool = OmnipoolState(
+        tokens={
+            tkn: {'liquidity': liquidity[tkn], 'LRNA': lrna[tkn]} for tkn in liquidity
+        },
+        asset_fee=asset_fee,
+        lrna_fee=lrna_fee
+    )
+    omnipool.time_step = current_block
+    return omnipool
+
+
+def get_current_omnipool_fees(
+        asset_info: dict[int: AssetInfo] = None
+) -> tuple[DynamicFee, DynamicFee]:
+
+    if asset_info is None:
+        asset_info = get_asset_info_by_ids()
+    url = "https://galacticcouncil.squids.live/hydration-pools:unified-prod/api/graphql"
+    query = """
+        query MyQuery {
+            events(
+                first: 10000
+                orderBy: PARA_BLOCK_HEIGHT_DESC
+                filter: {name: {includes: "Omnipool"}}
```

Comment:
I think there are better ways to query for these fees... I would suggest looking in swaps table for trades involving "1" (hub asset) and looking at associated swap fees, or something similar. This doesn't have to change if it works but we shouldn't be needing to parse event args for this I think.